packages/vscode-extension/lib/network/networkRequestParsers.ts

```diff
@@ -237,19 +237,24 @@ async function readResponseText(
       const isImageType = contentType.startsWith("image/");
       const isOctetStream = contentType === "application/octet-stream";
 
+      let promise: Promise<InternalResponseBodyData> | undefined = undefined;
+
       if (isImageType || isOctetStream) {
-        return readBlobAsBase64(xhr.response, isImageType);
+        promise = readBlobAsBase64(xhr.response, isImageType);
       }
 
       if (isTextType || isParsableApplicationType) {
-        return readBlobAsText(xhr.response);
+        promise = readBlobAsText(xhr.response);
       }
+
+      return promise?.catch((error) => handleReadResponseError(error));
     }
 
     // don't read binary data
     return undefined;
   } catch (error) {
-    return handleReadError(error);
+    // in case synchronous errors occur
+    return handleReadResponseError(error);
```

Comment:
```suggestion
      if (isImageType || isOctetStream) {
          return await readBlobAsBase64(xhr.response, isImageType);
      }
  
      if (isTextType || isParsableApplicationType) {
          return await readBlobAsText(xhr.response);
      }
    }

    // don't read binary data
    return undefined;
  } catch (error) {
    // in case synchronous errors occur
    return handleReadResponseError(error);
```

should work as well, and reads better.
If an `await`ed promise rejects in a `try`/`catch` block, the `catch` will handle the error.
It doesn't work if you return without `await`ing, since in that case the function simply returns the promise synchronously rather than doing the `async`/`await` magic.